src/FileSystem/SevenZipArchive.cpp

```diff
@@ -204,7 +204,7 @@ bool CSevenZipArchive::GetFile(unsigned int fid,
 			     &allocImp, &allocTempImp);
 	if (res == SZ_OK) {
 		buffer.resize(outSizeProcessed);
-		memcpy(&buffer[0], (char*)outBuffer + offset, outSizeProcessed);
+		memcpy(buffer.data(), (char*)outBuffer + offset, outSizeProcessed);
```

Comment:
I believe this isn't enough, per documentation:

- https://en.cppreference.com/w/cpp/container/vector/data: "If size() is 0, data() may or may not return a null pointer. "
- https://en.cppreference.com/w/cpp/string/byte/memcpy: "If either dest or src is an invalid or null pointer, the behavior is undefined, even if count is zero"

So it really needs a size > 0 check here before calling memcpy.

Reply:
fixed. let me know if you have any other suggestions